firefox-ios/Client/TabManagement/Tab.swift

```diff
@@ -405,6 +405,35 @@ class Tab: NSObject, ThemeApplicable {
 
     var profile: Profile
 
+    /// Returns true if this tab is considered inactive (has not been executed for more than a specific number of days).
+    /// Note: When `FasterInactiveTabsOverride` is enabled, tabs become inactive very quickly for testing purposes.
+    var isInactive: Bool {
+        let currentDate = Date()
+        let inactiveDate: Date
+
+        // Debug for inactive tabs to easily test in code
+        if UserDefaults.standard.bool(forKey: PrefsKeys.FasterInactiveTabsOverride) {
+            inactiveDate = Calendar.current.date(byAdding: .second, value: -10, to: currentDate) ?? Date()
+        } else {
+            inactiveDate = Calendar.current.date(byAdding: .day, value: -14, to: currentDate.noon) ?? Date()
+        }
+
+        // If we can't get a Tab timestamp, we consider it an active tab
+        guard let tabTimeStamp = lastExecutedTime ?? firstCreatedTime else {
+            return false
+        }
+
+        // If the tabDate is older than our inactive date cutoff, return true
+        let tabDate = Date.fromTimestamp(tabTimeStamp)
+        return tabDate <= inactiveDate
+    }
+
+    /// Returns true if this tab is considered active (has been executed within a specific numbers of days).
+    /// Note: When `FasterInactiveTabsOverride` is enabled, tabs become inactive very quickly for testing purposes.
+    var isActive: Bool {
+        return !isInactive
+    }
+
```

Comment:
I know most of this code is largely untested so it would be a big lift to test everything but could you add a few unit tests for this section at least?

Reply:
Absolutely! Good call out. I was literally thinking about this last night since I got the PR up right at EOD. 🤣 

I've filed [FXIOS-9987](https://mozilla-hub.atlassian.net/browse/FXIOS-9987) since I think Andy wants this fix in v131 beta.

Reply:
This refactor addresses FXIOS-9188 also, thanks @ih-codes 

(There's also an action item on that ticket to change the created & last executed timestamps to non-optionals but I can file a separate ticket for that if needed)

Reply:
Looks like I already had this up in [PR too](https://github.com/mozilla-mobile/firefox-ios/pull/21935) @mattreaganmozilla — apologies on the duplicated effort! And thanks for keeping track of these related tickets, I didn't realize we had some refactoring ones open for inactive tabs when Andy assigned me the bugs. 🙏